helix-core/src/test/java/org/apache/helix/manager/zk/TestZkHelixAdmin.java

```diff
@@ -560,6 +560,82 @@ public void testLegacyEnableDisablePartition() {
         2);
   }
 
+  @Test
+  public void testResetPartition() throws Exception {
+    String className = TestHelper.getTestClassName();
+    String methodName = TestHelper.getTestMethodName();
+    String clusterName = className + "_" + methodName;
+    String instanceName = "TestInstance";
+    String testResource = "TestResource";
+    System.out.println("START " + clusterName + " at " + new Date(System.currentTimeMillis()));
+    HelixAdmin admin = new ZKHelixAdmin(_gZkClient);
+    admin.addCluster(clusterName, true);
+    admin.addInstance(clusterName, new InstanceConfig(instanceName));
+    admin.enableInstance(clusterName, instanceName, true);
```

Comment:
Do we need a verifier check after this?

Reply:
Same as above. If the instance is not corrected added and enabled, we wont get all following expected exceptions.

Reply:
This is a unit test for sanity check. There is no running controller.  We have end-to-end tests for resetPartition() under webapp/TestResetPartitionState and integration/TestResetPartitionState.